app/scripts/loadout/dimLoadoutPopup.directive.js

```diff
@@ -134,6 +137,63 @@
       dimLoadoutService.applyLoadout(vm.store, loadout);
     };
 
+    // A dynamic loadout set up to level weapons and armor
+    vm.itemLevelingLoadout = function minBlueLoadout($event) {
+      var applicableItems = _.select(dimItemService.getItems(), function(i) {
+        return i.canBeEquippedBy(vm.store) &&
+          i.talentGrid && !i.talentGrid.xpComplete; // Still need XP
+      });
+
+      var bestItemFn = function(item) {
+        // Leave equipped items alone if they need XP
+        if (item.equipped) {
+          return 1000;
+        }
+
+        var value = 0;
+        // Prefer locked items (they're stuff you want to use/keep)
+        // and apply different rules to them.
+        if (item.locked) {
+          value += 500;
+          value += [
+            'Common',
+            'Uncommon',
+            'Rare',
+            'Legendary',
+            'Exotic'
+          ].indexOf(item.tier) * 10;
+        } else {
+          // For unlocked, prefer blue items so when you destroy them you get more mats.
+          value += [
+            'Common',
+            'Uncommon',
+            'Exotic',
+            'Legendary',
+            'Rare'
+          ].indexOf(item.tier) * 10;
+        }
+
+        // Choose the item w/ the highest XP
+        value += 10 * (1 - (item.talentGrid.totalXP / item.talentGrid.totalXPRequired));
```

Comment:
Shouldnt you use (item.talentGrid.totalXP / item.talentGrid.totalXPRequired)
rather than 1 - (item.talentGrid.totalXP / item.talentGrid.totalXPRequired)
This will cause it to prefer the less leveled items rather than finishing the almost lvled items.